webapp/channels/src/components/advanced_text_editor/remote_user_hour.tsx

```diff
@@ -0,0 +1,133 @@
+
+import {getChannel} from 'mattermost-redux/selectors/entities/channels';
+import {getTimezoneForUserProfile} from 'mattermost-redux/selectors/entities/timezone';
+import {getCurrentUserId, getUser, makeGetDisplayName} from 'mattermost-redux/selectors/entities/users';
+
+import Moon from 'components/common/svg_images_components/moon_svg';
+import Timestamp from 'components/timestamp';
+
+import Constants from 'utils/constants';
+
+import type {GlobalState} from 'types/store';
+
+const getDisplayName = makeGetDisplayName();
+
+const Container = styled.div`
+    display: flex;
+    aling-items: center;
+    padding: 8px 24px;
+    font-size: 12px;
+    color: rgba(var(--center-channel-color-rgb), 0.75);
+
+    & + .AdvancedTextEditor {
+        padding-top: 0;
+    }
+
+    .DoNotDisturbWarning + &{
+        display: none;
+    }
+
+    time {
+        font-weight: 600;
+    }
+`;
+
+const Icon = styled(Moon)`
+    svg {
+        width: 16px;
+        height: 16px;
+    }
+    svg path {
+        fill: rgba(var(--center-channel-color-rgb), 0.75);
+    }
+    margin: 0 2px;
+`;
+
+type Props = {
+    channelId: string;
+}
+
+const RemoteUserHour = ({channelId}: Props) => {
+    const userId = useSelector(getCurrentUserId);
+    const channel = useSelector((state: GlobalState) => getChannel(state, channelId));
+    const channelMembersIds = channel?.name.split('__');
+
+    let teammateId = '';
+    if (channelMembersIds && channelMembersIds.length === 2) {
+        teammateId = channelMembersIds[0];
+        if (teammateId === userId) {
+            teammateId = channelMembersIds[1];
+        }
+    }
+
+    const displayName = useSelector((state: GlobalState) => getDisplayName(state, teammateId, true));
+    const teammate = useSelector((state: GlobalState) => getUser(state, teammateId));
+    const [timestamp, setTimestamp] = useState(0);
+    const [showIt, setShowIt] = useState(false);
+
+    const teammateTimezone = getTimezoneForUserProfile(teammate);
+
+    useEffect(() => {
+        const teammateUserDate = DateTime.local().setZone(teammateTimezone.useAutomaticTimezone ? teammateTimezone.automaticTimezone : teammateTimezone.manualTimezone);
+        setTimestamp(teammateUserDate.toMillis());
+        setShowIt(teammateUserDate.get('hour') >= Constants.REMOTE_USERS_HOUR_LIMIT_END_OF_THE_DAY || teammateUserDate.get('hour') < Constants.REMOTE_USERS_HOUR_LIMIT_BEGINNING_OF_THE_DAY);
+
+        const interval = setInterval(() => {
+            const teammateUserDate = DateTime.local().setZone(teammateTimezone.useAutomaticTimezone ? teammateTimezone.automaticTimezone : teammateTimezone.manualTimezone);
+            setTimestamp(teammateUserDate.toMillis());
+            setShowIt(teammateUserDate.get('hour') >= Constants.REMOTE_USERS_HOUR_LIMIT_END_OF_THE_DAY || teammateUserDate.get('hour') < Constants.REMOTE_USERS_HOUR_LIMIT_BEGINNING_OF_THE_DAY);
+        }, 1000 * 60);
+        return () => clearInterval(interval);
```

Comment:
Q: if I open a DM where the conditions are met and the interval is set and then I click another DM where there conditions are not met. Will the interval be cleared in that case?

Reply:
The conditions here doesn't matter. Here we create the interval when the component is mount, if something changes in the dependency list, we destroy the interval and create a new one, and whenever the component is unmount, the interval gets destroyed. I think is consistent and behaving as expected.

Reply:
that makes sense 👍